src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useTradeExecution.tsx

```diff
@@ -208,6 +213,7 @@ export const useTradeExecution = (
       const stepBuyAssetAssetId = hop.buyAsset.assetId
 
       if (swapperName === SwapperName.CowSwap) {
+        if (!isExecutableTradeQuote(tradeQuote)) throw new Error('Unable to execute trade')
```

Comment:
can just do this single check at the top and then we are guaranteed to have the receive address for execute and don't need the extra runtime checks in these if conditions

Reply:
Added in 32e54ca544 - note not at the very very top of the function so we ensure all event handlers are still present